rplugin/python3/deoplete/deoplete.py

```diff
@@ -53,6 +55,13 @@ def __init__(self, vim):
         self.__profile_start = 0
 
     def completion_begin(self, context):
+        if not logger.init:
+            log_file = None
+            if self.__vim.vars['deoplete#enable_debug']:
+                self.debug_enabled = True
+                log_file = self.__vim.vars['deoplete#debug_log']
+            logger.setup(self.__vim, log_file)
```

Comment:
Why `logger.setup` is `g:deoplete#enable_debug` even false?

If
```diff
        if not logger.init:
            log_file = None
            if self.__vim.vars['deoplete#enable_debug']:
                self.debug_enabled = True
                log_file = self.__vim.vars['deoplete#debug_log']
-           logger.setup(self.__vim, log_file)
+               logger.setup(self.__vim, log_file)
```
It works the same.

Also, I think not necessary `root.addHandler(logging.NullHandler())`
https://github.com/Shougo/deoplete.nvim/pull/230/files#diff-fff55db87196854df902973d7b43a6dbR37